### DISCUSSION TITLE
Heizungsmischer 3 Wege

### DISCUSSION THREAD

**Discussion by meier08** [arendst/Tasmota#22688] [Q&A]
Hallo, zusammen.
Ich bin Neuling bei der Programmierung von ESP's.
Durch den Ausfall der Mischersteuerung meiner Heizung bin ich auf die Idee gekommen, das Ganze mit einem ESP 8266-01 in Verbindung mit einem Dual Kanal WIFI Relais Modul, geregelt von einem DS18x20 Temeraturfühler zu realisieren.
Leider fehlen mir sämtliche Grundlagen für dieses Projekt. Deshalb hoffe ich, hier jemanden zu finden, der mir bei der Entwicklung helfen kann oder evtl. selbst schon so ein Projekt verwirklicht hat.

Es soll eine, händisch veränderbare Solltemperatur (DS18x20) vorgegeben werden, auf die der MIschermotor hinarbeitet.
Ist die Temperatur zu niedrig soll der Mischer in 1 Sekunden Schritten öffnen (Relais 1), ist die Temperatur zu hoch, soll der Mischer in 1 Sekunden Schritten schließen (Relais 2).

Kann mir jemand dabei helfen.
Danke schon mal im Voraus

**Comment by spueli99** [comment]
Theoretisch sollte das funktionieren.
Zwar eine andere Baustelle aber funktioniert wohl. 
https://community.smarthome-for-dummies.de/viewtopic.php?t=1070

**Reply by meier08** [reply]
Danke für die schnelle Antwort.
Jedoch ist das Ganze für einen größeren ESP ausgelegt und hat auch 3 Temperaturfühler.
Ich habe nur diesen ESP: 
https://de.aliexpress.com/item/1005004404636309.html?spm=a2g0o.order_list.order_list_main.5.58895c5f7qtGIr&gatewayAdapt=glo2deu

Auch fühle ich mich als Anfänger nicht ganz in der Lage, das Programm auf meine Bedürfnisse abzuändern.

Was ich bereits habe:
- Den ESP auf Taxmota geflasht.
- Wlan eingerichtet
- Relais 1 auf GPIO 1
- Relais 2 auf GPIO 2

Den Temperaturfühler versuche ich im Laufe des Tages noch einzubinden.

Alles Andere ist leider noch offen.

**Comment by Noschvie** [comment]
Hallo,
In diesem Forum hier sollte Englisch bevorzugt verwendet werden…
Kurz meine Meinung dazu in D: das von dir verlinkte Board würde ich dafür nicht verwenden, sondern ein Board mit einem ESP32 und einer passenden Spannungsversorgung. Da gibt es einige, auch welche mit 220V oder 12-24V. Du könntest auch einen Shelly plus 2 (ohne PM!) und dem AddOn verwenden. Als Firmware jedenfalls Tasmota und einen PID Controller.

My brief opinion on this: I wouldn't use the board you linked to for this, but rather a board with an ESP32 and a suitable power supply. There are a few, including some with 220V or 12-24V. You could also use a Shelly plus 2 (without PM!) and the AddOn. In any case, the firmware should be Tasmota and a PID controller.

Does anyone have a link for a PID controller like https://tasmota.github.io/docs/PID-Control/

**Comment by Noschvie** [comment]
Hat der Mischer ein Potentiometer- oder Positionssensor, um die tatsächliche / aktuelle Position zu kennen?

**Reply by meier08** [reply]
Nein, der Mischer hat kein Potenziometer.
Wenn, dann kann die Position nur über die Laufzeit ermittelt werden.

**Comment by meier08** [comment]
Nein, der Mischer hat kein Potenziometer.
Wenn, dann kann die Position nur über die Laufzeit ermittelt werden.

**Comment by Noschvie** [comment]
Hello @stefanbode is it possible to use the "enhanced ESP32 shutter support" to control a mixer of a heating boiler?

The result of the PID controller should be taken as input for the mixer...

**Comment by krishy2** [comment]
I've did it. It's working. Let me know if you need help.

**Reply by meier08** [reply]
ich bekomme es leider nicht hin.

**Comment by Noschvie** [comment]
Hello @krishy2 I have no real use for it, but I am very interested out of technical curiosity.

**Comment by krishy2** [comment]
Any ESP device with at least 2 relays.
It is called 3 step PID in the industry.
Tasmota:
Compile your own firmware (there are some tricks).
Need a Present Value, like a thermometer. It could be local or could be via mqtt.
Need to do the pid tuning.

**Reply by Noschvie** [reply]
Which defines are needed for „self compiled“ and how to control both relays?

**Comment by krishy2** [comment]
2 relays: define a shutter.
So the output of PID in % dimension goes to the shutter.
define pid is necessary 
in xdrv pid there are the settings
timeprop must be explicitly turned off config as stated in xdrv. Otherwise strange behavior will be carried out.
temp (or other) sensor must be working. PID uses it by name, so if you do an alias on a ds18b20, then the alias needs to be used.
the pid parameters are precompiled, but using rules with on boot event the fine-tuned values can easily be retained.
All the pid parameters are not retain!
